### ISSUE TITLE
Detox fails with .mp4 file already exist

### ISSUE THREAD

**Issue by vincenzovitale** [wix/Detox#1147]
#### Description

We are getting:

detox[13524] ERROR: [ArtifactsManager.js/PLUGIN_ERROR] Caught exception inside function call: SimulatorLogPlugin.onIdleCallback()
  err: Error: EEXIST: file already exists, link '/private/var/folders/fq/z7qw6rbd0x70lxrqzpstl3n00000gn/T/830569cf-7445-4e9e-a172-cea1b88ee4
71.log' -> 'ios/build/detox-artifacts/✗

#### Steps to Reproduce
No idea, sometimes detox fails with this error. But not always.

#### Detox, Node, Device, Xcode and macOS Versions

* Detox: 10.0.5
* React Native: 0.57.2
* Node: 8.9.4
* Device: iPhone X simulator
* Xcode: 10.1
* macOS: 10.14.2

#### Device and verbose Detox logs

detox[13524] ERROR: [ArtifactsManager.js/PLUGIN_ERROR] Caught exception inside function call: SimulatorLogPlugin.onIdleCallback()
  err: Error: EEXIST: file already exists, link '/private/var/folders/fq/z7qw6rbd0x70lxrqzpstl3n00000gn/T/86188784-a0c0-4ba9-a10f-8524affba2
70.log' -> 'ios/build/detox-artifacts/✓ The XXX app _ios_ 1) Should have a XXX/test.log'
detox[13524] ERROR: [ArtifactsManager.js/PLUGIN_ERROR] Caught exception inside function call: SimulatorRecordVideoPlugin.onIdleCallback()
  err: Error: EEXIST: file already exists, link '/private/var/folders/fq/z7qw6rbd0x70lxrqzpstl3n00000gn/T/f540c0e4-a167-4315-bb46-069153deb064.mp4' -> 'ios/build/detox-artifacts/✓ The XXX app _ios_ 1) Should have XXX/test.mp4'
detox[13524] ERROR: [exec.js/SPAWN_STDERR, #9] Failed to register video stream: Error Domain=NSPOSIXErrorDomain Code=22 "Invalid argument" U
serInfo={connectionUUID=25380D0A-BE0B-44CF-926F-BD55FE99799E, message=Invalid connection UUID specified.}

....

detox[13524] ERROR: [ArtifactsManager.js/PLUGIN_ERROR] Caught exception inside function call: SimulatorLogPlugin.onIdleCallback()
  err: Error: EEXIST: file already exists, link '/private/var/folders/fq/z7qw6rbd0x70lxrqzpstl3n00000gn/T/830569cf-7445-4e9e-a172-cea1b88ee4
71.log' -> 'ios/build/detox-artifacts/✗ The XXX app _ios_ 2) Should enter and exit XXX/test.log'
detox[13524] ERROR: [exec.js/SPAWN_STDERR, #10] Failed to register video stream: Error Domain=NSPOSIXErrorDomain Code=22 "Invalid argument"

**Comment by noomorph** [comment]
Let's get it straight. I also noticed the behavior of Simulator deteriorated since XCode 10, and some command calls (which are timewise close to each other) lead to cryptic errors.

Such errors at times are hard to reproduce in a consistent way, though I did some attempts at stabilizing artifact plugins under increased load: https://github.com/wix/Detox/blob/master/detox/src/artifacts/log/ios/SimulatorLogPlugin.test.js

Currently, to my regret, I have no resources to dive into this issue so your help will be appreciated if you provide a test which reproduces `err: Error: EEXIST: file already exists, link` error at least in one of the artifact plugins.

### Last but not least, Detox does not fail on artifact plugin errors.

There is a try-catch statement in the code which logs the errors but it does not affect tests execution.
You are welcome to prove otherwise, and I'll be happy to fix it if there is some omission from our side.

In other words, the artifact recording workflow was designed to be as much isolated from the test runner workflow as it's possible, so you should give more evidence if that is the bug you want to report.

**Comment by vincenzovitale** [comment]
Hi, thanks for the fast reply!

I'll try to find a consistent way to reproduce the error. Indeed it looks like Xcode 10 is the culpable here and the fact tests are failing in a flaky way makes things complicate.
